### PR TITLE
Create IsValid

### PR DESCRIPTION
**Pull request by ian952**
This PR completes item 1 of #69 

Changelog:

1. Added GetBoardPiece which returns the piece at the given location
2. Added IsValid which checks if a move is valid
3. Added additional checking in IsValid of model/game.go to check if the given board dimension matches NumLines

Issue Link:

https://github.com/SEkiSoft/gogogo/issues/69

### CODE REVIEW

**Comment by Denton-L** [comment]
👍

File: model/move.go [hunk 1]
```diff
@@ -35,8 +35,15 @@ func (m *Move) PreSave() {
 }
 
 func (m *Move) IsValid(game *Game) *Error {
-	// TODO
-	return nil
+	current_piece, err := game.GetBoardPiece(m.X, m.Y)
```
Comment (DavidLu1997) [inline model/move.go]: minor nit, convention should be `currentPiece`

File: model/move.go [hunk 2]
```diff
@@ -35,8 +35,15 @@ func (m *Move) PreSave() {
...
+
+	if err != nil {
+		return err
+	} else if current_piece != 0 {
+		return NewLocError("Move.IsValid", "Spot is occupied", nil, "")
+	} else {
+		return nil
```
Comment (DavidLu1997) [inline model/move.go]: No `else-return`

**Comment by DavidLu1997** [comment]
+1

**Pull request merged**